src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -104,16 +105,14 @@ private void unlink(Node<T> node) {
 
     private Node<T> getNode(int index) {
         checkIndex(index);
-        Node<T> currentNode = head;
-        if (index < (size / 2)) {
-            for (int i = 0; i < index; i++) {
-                currentNode = currentNode.next;
-            }
-        } else {
-            for (int i = index; i > 0; i--) {
-                currentNode = currentNode.next;
-            }
+        boolean isFirstHalf = size / 2 >= index;
+        Node<T> currentNode = isFirstHalf ? head : tail;
+        int iteration = isFirstHalf ? index : size - index - 1;
+
+        for (int i = 0; i < iteration; i++) {
+            currentNode = isFirstHalf ? currentNode.next : currentNode.prev;
         }
+
```

Comment:
I'd suggest just making 2 methods: `findFromHead()` & `findFromTail()` and calling them based on the `isFirstHalf` with a bit fewer ternary operators to improve readability